### ISSUE TITLE
`no-magic-number` and composed constants

### ISSUE THREAD

**Issue by dinoboff** [eslint/eslint#6647]
**What version of ESLint are you using?**
v3.0.1

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
```yaml
extends: eslint:recommended
env:
  es6: true
  node: true
parserOptions: 
  ecmaVersion: 6
  sourceType: script
rules:
  no-magic-numbers: error
```
**What did you do? Please include the actual source code causing the issue.**
index.js
```js
'use strict';

const ms = 1;
const second = ms * 1000;
const minute = second * 60;
const hour = minute * 60;
const day = hour * 24;

Object.assign(module.exports, {ms, second, minute, hour, day});
```

```shell
eslint index.js
```
**What did you expect to happen?**
No error.

**What actually happened? Please include the actual, raw output from ESLint.**

    path/to/index.js
      4:21  error  No magic number: 1000  no-magic-numbers
      5:25  error  No magic number: 60    no-magic-numbers
      6:23  error  No magic number: 60    no-magic-numbers
      7:20  error  No magic number: 24    no-magic-numbers

    ✖ 4 problems (4 errors, 0 warnings)

**Comment by gyandeeps** [comment]
Technically speaking, rule is behaving correctly here. 1000, 60, 24 are magic numbers.

I think this is whats expected:

```js
'use strict';

const ms = 1;
const timeMultiplier = 60;
const secMultiplier = 1000;
const dayMultiplier = 24;

const second = ms * secMultiplier;
const minute = second * timeMultiplier;
const hour = minute * timeMultiplier;
const day = hour * dayMultiplier;

Object.assign(module.exports, {ms, second, minute, hour, day});
```

\* Change the variable names accordingly.

**Comment by dinoboff** [comment]
I don't think it improve readability.

I think adding a option to allow magic number in assignment would be useful.

**Comment by ilyavolodin** [comment]
This is an exact duplicate of #4226
This rule is supposed to be very strict. If it's doesn't fit your use-case, you should disable it.
Examples you provided might be readable, but here's an example of composed constant that wouldn't make sense to a lot of people:

```js
const mass = 40;
const energy = mass * (3 * 10 ** 8) ** 2;
```

Closing as duplicate of #4226

**Comment by mysticatea** [comment]
```js
const secondsPerMinute = 60;
const minutesPerHour = 60;
const hoursPerDay = 24;
const millisPerSecond = 1000;
const millisPerMinute = millisPerSecond * secondsPerMinute;
const millisPerHour = millisPerMinute * minutesPerHour;
const millisPerDay = millisPerHour * hoursPerDay;
```

**Comment by dinoboff** [comment]
@ilyavolodin sorry for the duplicate.

Magic number in array access or object properties assignment is quite loose.
